sample_project/Assets/SampleViewer/Samples/FeatureLayer/Scripts/FeatureLayerInputManager.cs

```diff
@@ -0,0 +1,200 @@
+// Copyright 2024 Esri.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at: http://www.apache.org/licenses/LICENSE-2.0
+//
+
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class FeatureLayerInputManager : MonoBehaviour
+{
+    public FeatureData FeatureData;
+
+    [SerializeField] private Transform container;
+    private InputActions inputActions;
+    private bool isLeftShiftPressed;
+    private List<GameObject> items = new List<GameObject>();
+    [SerializeField] private Material outlineMat;
+    [SerializeField] private GameObject properties;
+    [SerializeField] private GameObject propertiesView;
+    private TouchControls touchControls;
+    private FeatureLayerUIManager uiManager;
+
+    private IEnumerator AddItemsToScrollView()
+    {
+        foreach (var item in items)
+        {
+            SetScrollViewItems(item);
+            yield return new WaitForSeconds(0.01f);
+        }
+
+        StopCoroutine("AddItemsToScrollView");
+    }
+
+    private void Awake()
+    {
+#if !UNITY_IOS && !UNITY_ANDROID && !UNITY_VISIONOS
+        inputActions = new InputActions();
+#else
+        touchControls = new TouchControls();
+#endif
+    }
+
+    public void ClearAdditionalMaterial(GameObject feature)
+    {
+        Material[] materialsArray = new Material[feature.GetComponentInChildren<Renderer>().materials.Length - 1];
+
+        for (int i = 0; i < feature.GetComponentInChildren<Renderer>().materials.Length - 1; i++)
+        {
+            materialsArray[i] = feature.GetComponentInChildren<Renderer>().materials[i];
+        }
+
+        feature.GetComponentInChildren<Renderer>().materials = materialsArray;
```

Comment:
```suggestion
        renderer.materials = renderer.materials.SkipLast(1).ToArray();
```
In addition to Stephens feedback.